dashboard/app/controllers/api/v1/pd/applications_controller.rb

```diff
@@ -139,6 +139,7 @@ def update
         return render status: :bad_request, json: {errors: @application.errors.full_messages}
       end
 
+      @application.send_decision_email if status_changed && @application.should_send_decision_email?
```

Comment:
This logic used to be part of `log_status` which is called via a callback when the status changes
https://github.com/code-dot-org/code-dot-org/blob/6abe34f03bc66feffe8710d7ecf5661ce877cc08/dashboard/app/models/pd/application/teacher_application.rb#L87

But sending the email right away also triggers an update to the `status_timestamp_changelog` https://github.com/code-dot-org/code-dot-org/blob/6abe34f03bc66feffe8710d7ecf5661ce877cc08/dashboard/app/models/pd/application/application_base.rb#L164

which re-triggers a save, which re-triggers the call to log_status, which re-triggers the update to the `status_timestamp_changelog`, etc. Instead, put the sending of the email inside the controller.